### PR TITLE
made new types of literals

### PR DESCRIPTION
**Pull request by brennanjl**
Added the ability to detect several types of literals

### CODE REVIEW

File: SQLLexer.g4 [hunk 1]
```diff
@@ -126,11 +123,28 @@ IDENTIFIER:
...
+    'true'
+    | 'false'
+;
+
+NUMERIC_LITERAL:
+    [0-9]+
+;
+
+BLOB_LITERAL:
+    '0x' [0-9a-f]+
```
Comment (Yaiba) [inline SQLLexer.g4:138]: I kind prefer to use fragment defined below: so `'0x' HEX_DIGIT+`, same to the NUMERIC_LITERAL

File: SQLLexer.g4 [hunk 2]
```diff
@@ -126,11 +123,28 @@ IDENTIFIER:
     | [A-Z_] [A-Z_0-9]* // Ordinary identifiers
 ; // TODO check: needs more chars in set
 
-NUMERIC_LITERAL: ((DIGIT+ ('.' DIGIT*)?) | ('.' DIGIT+)) ('E' [-+]? DIGIT+)? | '0x' HEX_DIGIT+;
+// literals
```
Comment (Yaiba) [inline SQLLexer.g4:126]: Didn't realize floating point value was not removed 🤦

File: SQLLexer.g4 [hunk 3]
```diff
@@ -126,11 +123,28 @@ IDENTIFIER:
...
+NUMERIC_LITERAL:
+    [0-9]+
+;
+
+BLOB_LITERAL:
+    '0x' [0-9a-f]+
+;
 
-STRING_LITERAL: '\'' ( ~'\'' | '\'\'')* '\'';
+TEXT_LITERAL:
```
Comment (Yaiba) [inline SQLLexer.g4:141]: Why this is not the same as the old `STRING_LITERAL`?

File: SQLLexer.g4 (hunk 1 above)
Comment (brennanjl) [reply SQLLexer.g4:131]: Ok

Comment (brennanjl) [reply SQLLexer.g4:131]: Actually I sort've disagree. Hex digit is not used anywhere, it should just be removed IMO

File: SQLLexer.g4 (hunk 3 above)
Comment (brennanjl) [reply SQLLexer.g4:134]: Oh I changed this while debugging. Will change back

File: SQLLexer.g4 (hunk 1 above)
Comment (Yaiba) [reply SQLLexer.g4:131]: either is fun.  Maybe also delete NUMBERIC_LITERAL?

Comment (brennanjl) [reply SQLLexer.g4:131]: We use numeric literal

Comment (Yaiba) [reply SQLLexer.g4:131]: ah sorry, I mean `DIGIT`, you've deleted

**Review by Yaiba — APPROVED** [review]

**Pull request merged**